src/helper.jl

```diff
@@ -12,3 +12,42 @@ end
 macro maybe_threaded(ex)
     esc(_maybe_threaded(ex))
 end
+
+function _init_backend()
+    quote
+        using Pkg
+        deps = Pkg.project().dependencies
+        const backend = JACC.JACCPreferences.backend
+        if backend == "cuda"
+            if !haskey(deps, "CUDA")
+                Pkg.add("CUDA")
+                @info "Added CUDA (be careful about committing Project.toml)"
+            end
+            using CUDA
+            @info "CUDA backend loaded"
+
+        elseif backend == "amdgpu"
+            if !haskey(deps, "AMDGPU")
+                Pkg.add("AMDGPU")
+                @info "Added AMDGPU (be careful about committing Project.toml)"
+            end
+            using AMDGPU
+            @info "AMDGPU backend loaded"
+
+        elseif backend == "oneapi"
+            if !haskey(deps, "oneAPI")
+                Pkg.add("oneAPI")
+                @info "Added oneAPI (be careful about committing Project.toml)"
+            end
+            using oneAPI
+            @info "oneAPI backend loaded"
+
+        elseif backend == "threads"
+            @info "Threads backend loaded"
+        end
+    end
+end
+
+macro init_backend()
+    return esc(_init_backend())
```

Comment:
Hi @PhilipFackler looks great. It would be good for our future selfs and collaborators to put a comment for wrapping the function in a macro.

Reply:
Seems like that's pretty common practice.

Reply:
Right, I mean is this done to resolve the back end during precompilation? It's not clear why a macro and not a function.

Reply:
Oh, because you can't have a `using` statement inside a function